zk.go

```diff
@@ -41,32 +41,32 @@ type NoopLogger struct{}
 func (NoopLogger) Printf(format string, a ...interface{}) {
 }
 
-// Loads leader from ZK endpoint.
+// Retrieves current Aurora leader from ZK.
 func LeaderFromZK(cluster Cluster) (string, error) {
 
 	var err error
 	var zkurl string
 
 	duration := defaultBackoff.Duration
-	for i := 0; i < defaultBackoff.Steps; i++ {
-		if i != 0 {
-			adjusted := duration
-			if defaultBackoff.Jitter > 0.0 {
-				adjusted = Jitter(duration, defaultBackoff.Jitter)
-			}
-			fmt.Println(" sleeping for: ", adjusted)
-			time.Sleep(adjusted)
-			duration = time.Duration(float64(duration) * defaultBackoff.Factor)
-		}
-		if zkurl, err = leaderFromZK(cluster); err == nil {
+	for step := 0; step < defaultBackoff.Steps; step++ {
+
+		// Attempt to find leader
+		zkurl, err = leaderFromZK(cluster)
+		if err == nil {
 			return zkurl, err
 		}
-		if err != nil {
-			fmt.Println("error in LeaderFromZK: ", err)
+
+		// Backoff if we failed to determine leader
+		adjusted := duration
+		if defaultBackoff.Jitter > 0.0 {
+			adjusted = Jitter(duration, defaultBackoff.Jitter)
 		}
+		fmt.Printf("Error determining Aurora leader: %v; retrying in %v\n", err, adjusted)
+		time.Sleep(adjusted)
```

Comment:
Hmm, I am not sure you understood my concern. There is the _Jitter_ function that should prevent scores of clients hitting ZK at the same time by spreading the retries.

And there is the retry count which presumably is greater than 0, so DOS scenario should not develop.

You could increase the retry count by 1 to achieve the same effect (holding off clients from retrying too rapidly or too often) if that is your goal. OTOH with this extraneous sleep clients are forced to pause even when there is no chance a successful connection can be made.